src/test/gitSecrets.test.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { describe } from 'mocha'
+
+import { spawnSync } from 'child_process'
+import * as assert from 'assert'
+import { getProjectDir } from './testUtil'
+import * as path from 'path'
+import { platform } from 'os'
+
+/**
+ * secrets patterns are set in the project root, .git/config
+ */
+describe('Test git-secrets scan', function () {
+    let projectDir: string
+
+    function setAllowListPatterns() {
+        const allowListPatterns: string[] = []
+
+        allowListPatterns.forEach(pattern => {
+            spawnSync('git-secrets', ['--add', '--allowed', pattern], { cwd: projectDir })
+        })
+    }
+
+    function setDenyListPatterns() {
+        const denyListPatterns: string[] = []
+
+        denyListPatterns.forEach(pattern => {
+            spawnSync('git-secrets', ['--add', pattern], { cwd: projectDir })
+        })
+    }
+
+    before(function () {
+        if (platform() === 'win32') {
+            this.skip()
+        }
+        projectDir = path.join(getProjectDir(), '..', '..')
+        const result = spawnSync('cat', ['.git/config'], { cwd: projectDir })
+        console.log(`config ls: ${result.stdout.toString()}`)
+        spawnSync('git-secrets', ['--register-aws'], { cwd: projectDir })
+        setAllowListPatterns()
+        setDenyListPatterns()
+    })
+
+    it('ensures no git secrets are found', function () {
+        let result = spawnSync('which', ['git-secrets'], { cwd: projectDir })
```

Comment:
by doing the `curl` in this test, we can avoid hacks like `which`. We can download the script to an explicit location and call that location.